crates/subspace-service/src/lib.rs

```diff
@@ -248,6 +253,13 @@ where
 
     let select_chain = sc_consensus::LongestChain::new(backend.clone());
 
+    let bundle_collector = {
+        let confirmation_depth_k = get_chain_constants(client.as_ref())
+            .expect("Must always be able to get chain constants")
+            .confirmation_depth_k();
+        BundleCollector::new(client.clone(), confirmation_depth_k as usize)
```

Comment:
Why not invoke `get_chain_constants()` inside BundleCollector in order to simplify the construction here?